app.js

```diff
@@ -1,48 +1,149 @@
 
 mongoose
-  .connect('mongodb://localhost/auth-with-passport', {
+  .connect("mongodb://localhost/auth-with-passport", {
     useNewUrlParser: true,
     useUnifiedTopology: true,
-    useCreateIndex: true
+    useCreateIndex: true,
   })
-  .then(x => console.log(`Connected to Mongo! Database name: "${x.connections[0].name}"`))
-  .catch(err => console.error('Error connecting to mongo', err));
+  .then((x) =>
+    console.log(`Connected to Mongo! Database name: "${x.connections[0].name}"`)
+  )
+  .catch((err) => console.error("Error connecting to mongo", err));
 
-const app_name = require('./package.json').name;
-const debug = require('debug')(`${app_name}:${path.basename(__filename).split('.')[0]}`);
+const app_name = require("./package.json").name;
+const debug = require("debug")(
+  `${app_name}:${path.basename(__filename).split(".")[0]}`
+);
 
 const app = express();
 
 // Middleware Setup
-app.use(logger('dev'));
+app.use(logger("dev"));
 app.use(bodyParser.json());
 app.use(bodyParser.urlencoded({ extended: false }));
 app.use(cookieParser());
 
 // Express View engine setup
 
-app.set('views', path.join(__dirname, 'views'));
-app.set('view engine', 'hbs');
-app.use(express.static(path.join(__dirname, 'public')));
-app.use(favicon(path.join(__dirname, 'public', 'images', 'favicon.ico')));
+app.set("views", path.join(__dirname, "views"));
+app.set("view engine", "hbs");
+app.use(express.static(path.join(__dirname, "public")));
+app.use(favicon(path.join(__dirname, "public", "images", "favicon.ico")));
 
 // default value for title local
-app.locals.title = 'Express - Generated with IronGenerator';
+app.locals.title = "Express - Generated with IronGenerator";
+
+// passport:
+app.use(
+  session({
+    secret: "pLeAsE-EnTeR-sEcReT",
+    resave: true,
+    saveUninitialized: true,
+  })
+);
+
+passport.serializeUser((user, next) => {
+  next(null, user._id);
+});
+
+passport.deserializeUser((id, next) => {
+  User.findById(id)
+    .then((found) => next(null, found))
+    .catch((error) => {
+      console.log(error);
+      next();
+    });
+});
+
+app.use(flash());
+passport.use(
+  new LocalStrategy(
+    { passReqToCallback: true },
+    (request, username, password, next) => {
+      User.findOne({ username })
+        .then((user) => {
+          if (!user) {
+            return next(null, false, { message: "Incorrect username" });
+          }
+          if (!bcrypt.compareSync(password, user.password)) {
+            return next(null, false, { message: "Incorrect password" });
+          }
+          return next(null, user);
+        })
+        .catch((error) => {
+          console.log(error);
+          next();
+        });
+    }
+  )
+);
+
+// Bonus: Social Login
+// see lecture instructions
+const SlackStrategy = require("passport-slack").Strategy;
+
+passport.use(
+  new SlackStrategy(
+    {
+      clientID: "1012534720640.1091954637505",
+      clientSecret: "a10680ea274740d72bd0476a16eef92b",
```

Comment:
I haven't looked into the slack login strategy, but this client secret MAY need to be in the .env file.

Some companies can fine you for sharing the API secret on github. A student from the previous bootcamp got a fine of a few houndred 100 euros for sharing his api-secret (he didn't put the .env file in .gitignore). 
Just a fair warning;)